latest/index.bs

```diff
@@ -331,24 +317,917 @@ Conforming readers:
+translation transformation should be preferred to its equivalent affine. Input and output dimensionality MUST be
+identical and MUST equal the the length of the "translation" array (N). `translation` transformations are
+invertible.
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The path to a zarr-array containing the translation parameters.
+	The array at this path MUST be 1D, and its length MUST be `N`.</dd>
+  <dt><strong>url</strong></dt>
+  <dd>  An optional URL to the container in which the translation zarr-array is stored. If not provided,
+        the provided `path` MUST exist in this container.</dd>
+  <dt><strong>scale</strong></dt>
+  <dd> 	The scale parameters stored as a JSON list of numbers. The list MUST have length `N`.</dd>
+</dl>
+
+<div class=example>
+
+<pre class=include-code>
+path: examples/transformations/translation.json
+highlight: json
+</pre>
+
+defines the function:
+
+```
+x = i + 9 
+y = j - 1.42
+```
+</div>
+
+#### scale
+
+`scale` transformations are special cases of affine transformations. When possible, a scale transformation
+SHOULD be defined to its equivalent affine. Input and output dimensionality MUST be identical and MUST equal
+the the length of the "scale" array (N). Values in the `scale` array SHOULD be non-zero; in that case, `scale`
+transformations are invertible.
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The path to a zarr-array containing the scale parameters.
+	The array at this path MUST be 1D, and its length MUST be `N`.</dd>
+  <dt><strong>url</strong></dt>
+  <dd>  An optional URL to the container in which the scale zarr-array is stored. If not provided,
+        the provided `path` MUST exist in this container.</dd>
+  <dt><strong>scale</strong></dt>
+  <dd> 	The scale parameters stored as a JSON list of numbers. The list MUST have length `N`.</dd>
+</dl>
+
+<div class=example>
+
+<pre class=include-code>
+path: examples/transformations/scale.json
+highlight: json
+</pre>
+
+defines the function:
+
+```
+x = 3.12 * i
+y = 2 * j
+```
+</div>
+
+#### affine
+
+`affine` transformations from N-dimensional inputs to M-dimensional outputs are represented at `(N)x(M+1)`
+matrices in homogeneous coordinates. This transformation type is invertible when `N` equals `M`.
+The matrix may be stored as a 2D array or as a 1D array (row-major).
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The path to a zarr-array containing the affine parameters.
+	The array at this path MUST be 1D, its length MUST be `N*(M+1)`. </dd>
+  <dt><strong>url</strong></dt>
+  <dd>  An optional URL to the container in which the affine zarr-array is stored. If not provided,
+        the provided `path` MUST exist in this container.</dd>
+  <dt><strong>affine</strong></dt>
+  <dd> 	The affine parameters stored in JSON. The matrix may be stored
+        as a row-major flat list of numbers the list MUST be length `N*(M+1)`. </dd>
```

Comment:
Above it states that the matrix may be stored as 2D array or 1D array.

Reply:
good catch, thanks

Reply:
fixed by [be7d849](https://github.com/ome/ngff/pull/138/commits/be7d84982799bd9a99a66d3a75ea6abfd846af90)